thrift/server.cc

```diff
@@ -41,7 +41,14 @@
 #include <limits>
 #include <cctype>
 #include <vector>
+
+#ifdef THRIFT_USES_BOOST
+#include <boost/make_shared.hpp>
+#define make_shared_ns boost
+#else
```

Comment:
Don't use #defines (they follow different rules to ordinary C++ names), use `namespace alias = boost`. In this case I suggest `thrift_std` for `alias`.  The reasoning is we want to use std but are forced to use a substitute in older versions of boost.